ql/pricingengines/barrier/fdhestonbarrierengine.cpp

```diff
@@ -184,14 +185,13 @@ namespace QuantLib {
                     ext::dynamic_pointer_cast<StrikedTypePayoff>(
                                                             arguments_.payoff);
             // Calculate the vanilla option
-            ext::shared_ptr<DividendVanillaOption> vanillaOption(
-				ext::make_shared<DividendVanillaOption>(payoff,arguments_.exercise,
-                                          dividendCondition->dividendDates(), 
-                                          dividendCondition->dividends()));
+            auto vanillaOption =
```

Comment:
It might be slightly more efficient to allocate `vanillaOption` and `rebateOption` on the stack because they never leave the scope of this function.